### PR TITLE
Feature/AUT-120/support parsing figcaption element

### PR DESCRIPTION
**Pull request by pnal**
Related to: https://oat-sa.atlassian.net/browse/AUT-120

Depends on: 
- [x] https://github.com/oat-sa/tao-core/pull/3540
- [x] https://github.com/oat-sa/tao-item-runner-qti-fe/pull/289
- [x] https://github.com/oat-sa/extension-tao-testqti/pull/2272
- [x] https://github.com/oat-sa/lib-tao-qti/pull/117
- [x] https://github.com/oat-sa/qti-sdk/pull/313

<del>For local test need to use brach from qti-sdk, please run  `composer require "qtism/qtism:dev-feature/legacy/TR-1499/support_qti_figure_figcaption_elements as 0.25.2" -W`</del>

# How to test:

- create new passage
- add image to it
- add/edit image caption
- save
- check that it is rendered properly in preview, in items
- check export/re-import, delivery compilation, test runner

### CODE REVIEW

**Review by marpesia — CHANGES_REQUESTED** [review]
On Kitchen and Local environments, if I save the files, I have this error on the console. The file is saved correctly. I've tried with several images to check.
image

Update on the kitchen:
Now I can't save the Item
image

**Review by marpesia — CHANGES_REQUESTED** [review]
On Items and Assets
All AC are met ✅ 
Issue:
- After adding an image with a caption, I got several errors on clicking out of the Image and clicking on the A Block.
image
- On adding an Asset to an Item, the font size change affects the Item AND the Asset
image

**Review by marpesia — APPROVED** [review]
Awesome!

**Review Checklist**
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful
- [ ] Pull request's target is not `master`
- [ ] Commits are following [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [ ] Commits messages are meaningful
- [ ] Commits are atomic
- [ ] Changelog is updated according to changes (if applicable)
- [ ] Documentation is updated according to changes (if applicable)

**Review by Silvia-loza — APPROVED** [review]
It works well and looks very good.  
Responsiveness from typing text till you see it in figcaption is super fast as well. Great job!   :clap: :clap:
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [X] New code is respecting code style rules
- [X] New code is respecting best practices
- [X]  Pull request's target is not` master`
- [X] Feature is working correctly on my local machine (if applicable)
- [X] Acceptance criteria are respected
- [X] Pull request title and description are meaningful

File: model/qti/container/ContainerGap.php [hunk 1]
```diff
@@ -53,7 +51,8 @@ public function getValidElementTypes()
             'oat\\taoQtiItem\\model\\qti\\Tooltip',
             'oat\\taoQtiItem\\model\\qti\\feedback\\Feedback',
             \oat\taoQtiItem\model\qti\QtiObject::class,
-            'oat\\taoQtiItem\\model\\qti\\choice\\Gap'
+            'oat\\taoQtiItem\\model\\qti\\choice\\Gap',
+            \oat\taoQtiItem\model\qti\Figure::class,
```
Comment (andreluizmachado) [inline model/qti/container/ContainerGap.php:55]: can we use `{classname}::class` in all of them?
please check the other classes that you touched as well

File: model/qti/container/ContainerFigure.php [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+class ContainerFigure extends ContainerStatic
+{
+
+    /**
+     * return the list of available element classes
+     *
+     * @access public
+     * @return array
+     */
+    public function getValidElementTypes()
```
Comment (andreluizmachado) [inline model/qti/container/ContainerFigure.php:33]: can we use a return type hint?

File: model/qti/ParserFactory.php [hunk 3]
```diff
@@ -523,6 +542,14 @@ protected function getXIncludeNamespace()
         return $this->findNamespace('XInclude');
     }
 
+    protected function getHTML5Namespace()
```
Comment (andreluizmachado) [inline model/qti/ParserFactory.php:545]: can we use return type hint?

File: model/qti/ParserFactory.php [hunk 4]
```diff
@@ -1442,6 +1469,15 @@ private function buildImg(DOMElement $data)
         return $returnValue;
     }
 
+    private function buildFigCaption(DOMElement $data)
```
Comment (andreluizmachado) [inline model/qti/ParserFactory.php:1472]: can we use return type hint?

File: model/qti/ParserFactory.php [hunk 5]
```diff
@@ -1491,6 +1527,16 @@ private function buildTable(DOMElement $data)
         return $table;
     }
 
+    private function buildFigure(DOMElement $data)
```
Comment (andreluizmachado) [inline model/qti/ParserFactory.php:1530]: same here

**Review by andreluizmachado — CHANGES_REQUESTED** [review]

**Review by andreluizmachado — APPROVED** [review]
- [x] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [ ] Changelog is updated according to changes (if applicable)
- [ ] Documentation is updated according to changes (if applicable)

**Pull request merged**